packages/web-components/@sfx/autocomplete/src/autocomplete.ts

```diff
@@ -95,11 +166,80 @@ export default class Autocomplete extends LitElement {
     this.dispatchEvent(sentEvent);
   }
 
+  /**
+   * Increments the `selectedIndex` property by 1.
+   * If incrementing `selectedIndex` will cause it to be out of bounds,
+   * its value will be set to `0` instead.
+   */
+  selectNext(): void {
+    const normalizedSelectedIndex = Math.min(Math.max(-1, this.selectedIndex), this.itemCount - 1);
+    this.selectedIndex = (normalizedSelectedIndex + 1) % this.itemCount;
+  }
+
+  /**
+   * Decrements the `selectedIndex` property by 1.
+   * If decrementing `selectedIndex` will cause it to be out of bounds,
+   * its value will be set to the index of the last item instead.
+   */
+  selectPrevious(): void {
+    const normalizedSelectedIndex = Math.min(Math.max(0, this.selectedIndex), this.itemCount);
+    this.selectedIndex = (normalizedSelectedIndex - 1 + this.itemCount) % this.itemCount;
+  }
+
+  /**
+   * Returns a callback that sets [[selectedIndex]] to the given index.
+   *
+   * @param index The index to set.
+   * @returns A callback that sets [[selectedIndex]].
+   */
+  getSetSelectedIndexCallback(index: number): () => void {
```

Comment:
@jmbtutor I wonder if this `getSetSelectedIndexCallback` method  could be better named. The name is a bit confusing to me, I get by definition you are doing both but I'm not sure if in that case it should be something like `handleSelectedIndexCallback` or follow some other naming convention? What do you think?

Reply:
We also have started to use the get and set keywords elsewhere in this file and I know there are benefits to that, is this a method that would be better broken out into a get and a set method?

Reply:
We were hoping someone would flag this, because we didn't like the name very much, either. This function was so named because it returns a callback that sets the selected index.

> `handleSelectedIndexCallback`

This doesn't work because we're creating a closure that returns a callback. This function doesn't do any handling itself.

> is this a method that would be better broken out into a get and a set method?

No. Because we're creating a closure with a variable, we need to be able to accept that variable, which means it needs to be a function. For this particular name, there is nothing to set to that name (i.e. one is not expected to do anything like `this.getSetSelectedIndexCallback = /*...*/`), so a setter would not be appropriate. For setting `selectedIndex` inside, a setter would be pointless and irrelevant because it is simply a property that holds a value (that LitElement actually turns it into a setter is beside the point).

`createSelectedIndexSetter` is the best alternative I can think of. Thoughts?